app/service/auth_svc.py

```diff
@@ -88,21 +96,40 @@ async def login_user(self, request):
         :param request:
         :return: the response/location of where the user is trying to navigate
         """
-        data = await request.post()
-        username = data.get('username')
-        password = data.get('password')
-        if self.ldap_config:
-            verified = await self._ldap_login(username, password)
-        else:
-            verified = await self._check_credentials(request.app.user_map, username, password)
-
-        if verified:
-            self.log.debug('%s logging in:' % username)
-            response = web.HTTPFound('/')
-            await remember(request, response, username)
-            raise response
-        self.log.debug('%s failed login attempt: ' % username)
-        raise web.HTTPFound('/login')
+        try:
+            self.log.debug('Using login handler "%s" for login' % self._login_handler.name)
+            await self._login_handler.handle_login(request)
+        except web.HTTPRedirection as http_redirect:
+            raise http_redirect
+        except web.HTTPUnauthorized as http_unauthorized:
+            raise http_unauthorized
+        except Exception as e:
+            self.log.error('Exception when handling login request: %s' % e)
```

Comment:
Use `.exception(...)` and you'll get the stack trace and exception message along with whatever your message is.